src/stacks-hierarchy/Root_PurchaseHarborSearchScreen/HarborResults.tsx

```diff
@@ -0,0 +1,165 @@
+  searchText?: string;
+  fromHarborName?: string;
+};
+
+export const HarborResults: React.FC<Props> = ({
+  harbors,
+  onSelect,
+  searchText = '',
+  fromHarborName,
+}) => {
+  const styles = useThemeStyles();
+  const {t} = useTranslation();
+
+  const [harborResults, setHarborResults] = useState<StopPlaces | []>([]);
+  const {location} = useGeolocationState();
+  const currentLocation = location || undefined;
+
+  useEffect(() => {
+    harbors && setHarborResults(sortHarbors(harbors, currentLocation));
+  }, [harbors, currentLocation]);
+
+  useEffect(() => {
+    if (searchText.length > 1 && harbors) {
+      setHarborResults(
+        sortHarbors(harbors, currentLocation).filter((harbor) => {
+          return harbor.name.toLowerCase().includes(searchText.toLowerCase());
+        }),
+      );
+    } else {
+      harbors && setHarborResults(sortHarbors(harbors, currentLocation));
+    }
+  }, [searchText]);
+  const showEmptyResultText = !harborResults.length && !!searchText;
+
+  return (
+    <>
+      <View accessibilityRole="header" style={styles.subHeader}>
+        {showEmptyResultText ? (
+          <MessageBox
+            type="info"
+            message={t(HarborSearchTexts.messages.emptyResult)}
+          />
+        ) : (
+          <ThemeText type="body__secondary" color="secondary">
+            {getheading(t, searchText, fromHarborName, currentLocation)}
+          </ThemeText>
+        )}
+      </View>
+      <View>
+        {harborResults.map((harbor, index) => (
+          <View style={styles.rowContainer} key={harbor.id}>
+            <View style={styles.buttonContainer}>
+              <TouchableOpacity
+                accessible={true}
+                accessibilityLabel={
+                  t(HarborSearchTexts.results.item.a11yLabel(harbor.name)) +
+                  screenReaderPause
+                }
+                accessibilityHint={t(HarborSearchTexts.results.item.a11yHint)}
+                accessibilityRole="button"
+                hitSlop={insets.symmetric(8, 1)}
+                onPress={() => onSelect(harbor)}
+                style={styles.button}
+                testID={'harborResult' + index}
+              >
+                <ThemeIcon svg={Boat} />
+                <View style={styles.nameContainer}>
+                  <ThemeText type={'body__primary--bold'}>
+                    {harbor.name}
+                  </ThemeText>
+                </View>
+              </TouchableOpacity>
+            </View>
+          </View>
+        ))}
+      </View>
+    </>
+  );
+};
+// sort by distance or alphabetically
+function sortHarbors(harbors: StopPlaces, location?: GeoLocation): StopPlaces {
+  if (location) {
+    return harbors
+      ?.map((stopPlace) => {
+        return {
+          id: stopPlace.id,
+          name: stopPlace.name,
+          distance: getDistance(stopPlace, location),
+        };
+      })
+      .filter((stopPlace) => stopPlace.distance != -1)
```

Comment:
Here we filter out stop places which hasn't `longitude` or `latitude` set. If we want that to happen, I think we rather should do it in the bff. If we don't want it to happen, we can either just show them last, or fallback to alphabetical if some of the stop places don't have coordinates.